src/google/protobuf/compiler/csharp/csharp_generator.cc

```diff
@@ -81,6 +81,13 @@ bool Generator::Generate(
     } else if (options[i].first == "base_namespace") {
       cli_options.base_namespace = options[i].second;
       cli_options.generate_directories = true;
+    } else if(options[i].first == "internal_access") {
+      std::string value = options[i].second;
+      if(value == "false" || value == "no" || value == "") {
```

Comment:
Looking at options in other languages, most boolean flags don't seem to support false/no. I'd *at least* suggest that "" should mean "true", but I'd be tempted to just have:

```
else if (options[i].first == "internal_access") {
  cli_options.internal_access = true;
}
```

(Note the space after the `if` by the way, for consistency of spacing.)